source/elements/oneMKL/source/architecture/other_architecture.inc.rst

```diff
@@ -11,10 +11,40 @@ This section covers all other features in the design of oneMKL architecture.
 
 .. _onemkl_spec_current_version:
 
-Current Version of this oneMKL Specification
+Specification Version and Compliance
 +++++++++++++++++++++++++++++++++++++++++++++
 
-This is the oneMKL specification which is part of the oneAPI specification version 1.0.0.
+Each oneMKL domain must define a preprocessor macro to represent the version of the specification that the implementation is compliant with.
+
+The macros for each domain are listed as follows:
+
+  | ONEMKL_BLAS_SPEC_VERSION
+  | ONEMKL_LAPACK_SPEC_VERSION
+  | ONEMKL_SPARSE_SPEC_VERSION
```

Comment:
can we use "SPBLAS" instead of "SPARSE" as someday there will also be SPSOLVERS  sparse solvers and the impls may not be in perfect lock step with SPBLAS ...  but will attempt to be interoperable

Reply:
@spencerpatty To use SPBLAS, we need to define SPBLAS as a domain in the spec's architecture section first, then make the corresponding change here. For now I cannot justify the divergence of the macro name from the domains defined in the spec here: https://oneapi-spec.uxlfoundation.org/specifications/oneapi/v1.3-rev-1/elements/onemkl/source/architecture/architecture#onemkl-namespaces

If an implementation is not compliant with the spec, then any macro name can be used by that implementation. If your suggestion is to make that change, I can create a new issue, but it would be out of scope for this RFC.

Reply:
so SPBLAS is a domain and it uses the "sparse" namespace.  Some day in the future there will also be a SPSOLVER  or sparse_solver domain introduced and it will also use the "sparse" namespace ...  I don't see the problem unless you are equating namespace with domain

Reply:
@spencerpatty When we add multiple sparse-related functionality, I think it makes sense to make the namespaces more clear, and if required, distinct for each Sparse group of APIs/algorithms, like SPBLAS, SPSOLVER etc.
I would like to refer to the point 5 of the https://github.com/mkrainiuk/oneMKL/blob/rfcs/README.md#rfc-ratification-process which requires any contributor to propose changes to oneAPI spec first, then make those changes in your backend implementation, if you want to comply with the oneAPI specification.

If you think having SPBLAS is better, then please feel free to open a new GH issue for the same to be reflected in all places in the oneAPI specification. Changing existing terminologies for domains and namespaces is out-of-scope for this RFC.

Reply:
namespaces won't be changing, but the name of the domains are different and they would share a namespace ... The name of the domain is not SPARSE but SPBLAS,  in some future time the domain of SPSOLVERS would also be added, but both are separate domains that will share a namespace.  I'm not sure why we need to have a GH issue ...  

https://github.com/uxlfoundation/oneAPI-spec/tree/main/source/elements/oneMKL/source/domains/spblas  the folder name of domain is "spblas" and the namespace is "sparse".  The name of the domain is Sparse BLAS here: 
 https://github.com/uxlfoundation/oneAPI-spec/blob/main/source/elements/oneMKL/source/domains/spblas/spblas.rst of which "SPBLAS" is short.
 
 On the other side, there is this page about "sparse linear algebra" https://github.com/uxlfoundation/oneAPI-spec/blob/main/source/elements/oneMKL/source/domains/sparse_linear_algebra.inc.rst which points to spblas.rst as the current only domain in sparse linear algebra, but again this is similar to the "dense linear algebra" page containing BLAS and LAPACK domains in it ...  and at some point sparse solvers will be part of the sparse linear algebra set as well.  
 
 All that being said, to me "SPBLAS" should be considered as the default here, and "SPARSE" would be in my mind deviating from existing documentation and specification and require github issue and PR changes...

Reply:
I updated the RFC to reflect SPBLAS as the macro name, and we now have some more work to clarify what we want to do with the sparse functionality in the spec itself: https://github.com/uxlfoundation/oneAPI-spec/issues/581

This hopefully unblocks this proposal from that confusion.